services/api/service.go

```diff
@@ -493,6 +527,84 @@ func (api *RelayAPI) startValidatorRegistrationDBProcessor() {
 	}
 }
 
+// simulateBlock sends a request for a block simulation to blockSimRateLimiter.
+func (api *RelayAPI) simulateBlock(ctx context.Context, opts blockSimOptions) (error, error) {
```

Comment:
let's make it explicit what the returned errors are!

```suggestion
func (api *RelayAPI) simulateBlock(ctx context.Context, opts blockSimOptions) (requestErr, validationErr error) {
```